python3Packages.kfactory: 1.14.4 -> 2.0.0

GaetanLepage opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

Diff: https://github.com/gdsfactory/kfactory/compare/v1.14.4...v2.0.0
Changelog: https://github.com/gdsfactory/kfactory/blob/v2.0.0/CHANGELOG.md

cc @fedeinthemix

- Built on platform:
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

GaetanLepage committed:
python3Packages.kfactory: 1.14.4 -> 2.0.0

Diff: https://github.com/gdsfactory/kfactory/compare/v1.14.4...v2.0.0

Changelog: https://github.com/gdsfactory/kfactory/blob/v2.0.0/CHANGELOG.md

fedeinthemix reviewed (commented):
`kfactory` should be updated in sync with `gdsfactor`, as the latter has a dependency on the former with tight version requirements.

GaetanLepage committed:
python3Packages.gdsfactory: 9.18.1 -> 9.20.1

Diff: https://github.com/gdsfactory/gdsfactory/compare/v9.18.1...v9.20.1

Changelog: https://github.com/gdsfactory/gdsfactory/blob/v9.20.1/CHANGELOG.md

GaetanLepage commented:
> `kfactory` should be updated in sync with `gdsfactor`, as the latter has a dependency on the former with tight version requirements.

Done.
But most `kfactory` tests are failing with:
```
tests/conftest.py:42: in lazy_datadir
    assert kf.config.project_dir is not None
E   AssertionError: assert None is not None
E    +  where None = Settings(n_threads=64, logfilter=LogFilter(level=<LogLevel.INFO: 'INFO'>, regex=None), display_type='image', meta_form...debug_names=False, write_cell_properties=True, write_context_info=True, write_file_properties=True, show_function=None).project_dir
E    +    where Settings(n_threads=64, logfilter=LogFilter(level=<LogLevel.INFO: 'INFO'>, regex=None), display_type='image', meta_form...debug_names=False, write_cell_properties=True, write_context_info=True, write_file_properties=True, show_function=None) = kf.config
```
